pkg/k8sutil/k8sutil.go

```diff
@@ -114,6 +136,80 @@ func NewClusterConfig(host string, tlsInsecure bool, tlsConfig *rest.TLSClientCo
 	return cfg, nil
 }
 
+func (c CRConfig) Runnable(ctx context.Context, host string, tlsInsecure bool, tlsConfig *rest.TLSClientConfig, logger log.Logger) (bool, error) {
+	cfg, err := NewClusterConfig(host, tlsInsecure, tlsConfig)
+	if err != nil {
+		return false, errors.Wrap(err, "instantiating cluster config failed")
+	}
+
+	kclient, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return false, errors.Wrap(err, "instantiating kubernetes client failed")
+	}
+
+	crdInstalled, err := c.IsCRDInstalled(kclient)
+	if !crdInstalled {
+		return false, err
+	}
+
+	missingPermissions, err := c.GetMissingPermissions(ctx, kclient)
+	if err != nil {
+		return false, err
+	}
+	if len(missingPermissions) > 0 {
+		return false, errors.Wrapf(err, "controller disabled because it lacks the required permissions: missingpermissions %v", missingPermissions)
+	}
+
+	return true, nil
+}
+
+func (c CRConfig) IsCRDInstalled(kclient *kubernetes.Clientset) (bool, error) {
```

Comment:
I don't think we want to expose  `IsCRDInstalled` and `GetMissingPermissions` outside of the package no? At least for now I don't see them being used outside

Reply:
Ya right. I kept it that way for ScrapeConfig validation. But we can enable later if needed